### ISSUE TITLE
Local vs global: \fontspec_set_family:Nnn etc

### ISSUE THREAD

**Issue by wspr** [wspr/fontspec#316]
Should functions like `\fontspec_set_family:Nnn` be always local, always global, or have `gset` variants?

**Comment by eg9** [comment]
If you say
```
\font\foo=cmr17
```
in legacy TeX, the font is loaded globally, so a new declaration for it will not load it again; but the assignment to the font selection command will respect grouping. Since
```
\global\font\foo=cmr17
```
is legal and will assign the font selection command globally. So perhaps both local and global variants could be provided, although I'd deprecate defining font families inside `document`.

Anyhow, if you decide that the function acts globally, its name should be with `gset` to make the action clear.

**Comment by wspr** [comment]
Yes, it’s not entirely clear at first look which is needed. I wanted to do some checks with plain NFSS and see what’s most natural to fit in with that (but ran out of time, hence the issue to remind me).

If I went with “global only” I’d probably change the name to \fontspec_new_family:Nnn with a check.

**Comment by wspr** [comment]
I definitely need to provide both variants.
